File: java/Array/removeduplicatesfromsortedarray.java
import java.util.*;
public class removeduplicatesfromsortedarray
{
    static private int removeDuplicates(int[] nums, int n)
    {
        if(n==0)
        {
            return 0;
        }
        // HashSet<Integer>s = new HashSet<>();
        // for(int i=0;i<n;i++)
        // {
        //     s.add(nums[i]);
        // }
        // int i=0;
        // for(int x:s)
        // {
        //     nums[i++]=x;
        // }
        // return s.size();
        int j=0;
        for(int i=1;i<n;i++)
        {
            if(nums[i]!=nums[j])
            {
                j++;
                nums[j]=nums[i];
            }
        }
        return j+1;
    }

    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int []arr = new int[n];
        for(int i=0;i<n;i++)
        {
            arr[i]=sc.nextInt();
        }
        int ans = removeDuplicates(arr,n);
        System.out.println(ans);
        for (int i = 0; i < ans; i++) {
            System.out.print(arr[i] + " ");
        }
        sc.close();
    }
}

// TC=O(n*log(n))+O(n) SC=O(n)
// TC=O(n) SC=O(1)

File: python/Array/removeduplicatesfromsortedarray.py
def removeDuplicates(arr,n):
    if n<2:
        return n
    # st=set()
    # for i in range(n):
    #     st.add(arr[i])
    # k=len(st)
    # j=0
    # for x in st:
    #     arr[j]=x
    #     j+=1
    # return k
    j=0
    for i in range(1,n):
        if arr[i]!=arr[j]:
            j+=1
            arr[j]=arr[i]
    return j+1
    
if __name__ == '__main__':
    n=int(input())
    arr=list(map(int, input().split()))
    length=removeDuplicates(arr,n)
    for i in range(length):
        print(arr[i], end=" ")
